src/components/DonationWidget/__tests__/index.spec.tsx

```diff
@@ -23,6 +22,21 @@ const billingInformation = {
   country: faker.address.country()
 };
 const donationAmount = faker.random.number(100);
+const donationResponse = {
+  status: 'succeeded',
+  message: `Your ${donationAmount} donation has been successfully processed`
+};
+const sendDonationSpy = jest.spyOn(HTTPService, 'sendDonation');
+
+beforeAll(() => {
+  global.fetch = jest.fn().mockResolvedValue({
+    json: jest.fn().mockResolvedValue(donationResponse)
```

Comment:
What happens if it fails? may It be worth it to add some basic error handling?

Reply:
Yes, right now the error handling happen thanks to the machine.
I would do that when we add proper error handling in terms of offering feedback to the user.
_But for now is like... I know that the machine has built-in support for when the promise fail enters to failure state._ Does it makes sense?